tests/integration/s2n_handshake_test_old_s_client.py

```diff
@@ -0,0 +1,363 @@
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+"""
+Handshake tests using Openssl 0.9.8 s_client against s2nd
+"""
+
+import argparse
+import os
+import sys
+import subprocess
+import itertools
+import multiprocessing
+import threading
+import uuid
+import re
+import string
+from os import environ
+from multiprocessing.pool import ThreadPool
+from s2n_test_constants import *
+from time import sleep
+
+S_CLIENT_NEGOTIATED_CIPHER_PREFIX="Cipher    : "
+
+PROTO_VERS_TO_S_CLIENT_ARG = {
+    S2N_TLS10 : "-tls1",
+    S2N_TLS11 : "-tls1_1",
+    S2N_TLS12 : "-tls1_2",
+}
+
+use_corked_io=False
+
+def cleanup_processes(*processes):
+    for p in processes:
+        p.kill()
+        p.wait()
+
+def validate_data_transfer(expected_data, s_client_out, s2nd_out):
+    """
+    Verify that the application data written between s_client and s2nd is encrypted and decrypted successfuly.
+    """
+    found = 0
+
+    for line in s2nd_out.splitlines():
+        if expected_data in line:
+            found = 1
+            break
+
+    if found == 0:
+        print ("Did not find " + expected_data + " in output from s2nd")
+        return -1
+
+    found = 0
+    for line in s_client_out.splitlines():
+        if expected_data in line:
+            found = 1
+            break
+
+    if found == 0:
+        print ("Did not find " + expected_data + " in output from s_client")
+        return -1
+
+    return 0
+
+def find_expected_cipher(expected_cipher, s_client_out):
+    """
+    Make sure s_client and s2nd negotiate the cipher suite we expect
+    """
+    s_client_out_len = len(s_client_out)
+    full_expected_string = S_CLIENT_NEGOTIATED_CIPHER_PREFIX + expected_cipher
+    for line in s_client_out.splitlines():
+        if full_expected_string in line:
+            return 0
+            break
+    print("Failed to find " + expected_cipher + " in s_client output")
+    return -1
+
+def read_process_output_until(process, marker):
+    output = ""
+
+    while True:
+        line = process.stdout.readline().decode("utf-8")
+        output += line
+        if marker in line:
+            return output
+
+    return output
+
+def try_handshake(endpoint, port, cipher, ssl_version, server_name=None, strict_hostname=False, server_cert=None, server_key=None,
+        server_cert_key_list=None, expected_server_cert=None, server_cipher_pref=None, ocsp=None, sig_algs=None, curves=None, resume=False, no_ticket=False,
```

Comment:
The code copying here does not bother me too much as we expect the tests in this file to be minimal and it's trivial later to remove these extra arguments.